materialize-snowflake/sqlgen.go

```diff
@@ -112,14 +112,16 @@ var snowflakeDialect = func(configSchema string) sql.Dialect {
 		Identifierer: sql.IdentifierFn(sql.JoinTransform(".",
 			sql.PassThroughTransform(
 				isSimpleIdentifier,
-				sql.QuoteTransform("\"", "\\\""),
+				sql.QuoteTransform(`"`, `""`),
 			))),
 		Literaler: sql.LiteralFn(sql.QuoteTransform("'", "''")),
 		Placeholderer: sql.PlaceholderFn(func(_ int) string {
 			return "?"
 		}),
-		TypeMapper:      mapper,
-		ColumnValidator: columnValidator,
+		TypeMapper:             mapper,
+		ColumnValidator:        columnValidator,
+		MaxColumnCharLength:    255,
+		CaseInsensitiveColumns: false,
```

Comment:
I'm under the impression that in Snowflake unless the identifier is quoted, the fields will be case-insensitive. Does that not mean we should set this to `true`?

Reply:
`CaseInsensitiveColumns` (which is a proxy for `caseInsensitiveFields` in the `materialize-boilerplate` validator) means that even if the fields are quoted, the destination system will still not allow more than one column with the same name that differs only in capitalization. Snowflake does allow `"thisField"` vs `"thisfield"`, so having it be `false` represents that. Admittedly I'm finding it hard to come up with a variable name that succinctly represents what this toggle does, so let me know if you have suggestions on something that would be more clear.